File: Programmers/Lv.1/달리기 경주/178871.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        String[] answer = players;
        
        Map<String, Integer> rankMap = new HashMap<>();
        for (int i = 0; i < players.length; i++) {
            rankMap.put(players[i], i);
        }
        
        for (String calling : callings) {
            int callingRank = rankMap.get(calling);
            String tempPlayer = players[callingRank - 1];
            
            rankMap.put(calling, callingRank - 1);
            rankMap.put(tempPlayer, callingRank);
            
            players[callingRank] = tempPlayer;
            players[callingRank - 1] = calling;
        }
        
        return answer;
    }
}

File: Programmers/Lv.1/달리기 경주/178871.js
function solution(players, callings) {
    const answer = players;
    
    const rankObj = {};
    for (let i = 0; i < players.length; i++) {
        rankObj[players[i]] = i;
    }
    
    callings.forEach(calling => {
        const callingRank = rankObj[calling];
        const tempPlayer = players[callingRank - 1];
        
        rankObj[calling] = callingRank - 1;
        rankObj[tempPlayer] = callingRank;
        
        players[callingRank] = tempPlayer;
        players[callingRank - 1] = calling;
    });
    
    return answer;
}
